### DISCUSSION TITLE
Sampling with fixed repetitions - randomisation not working

### DISCUSSION THREAD

**Discussion by fin329** [jspsych/jsPsych#2708] [Q&A] [answered]
Hey everyone,

I'm currently trying to randomise my trials with the sampling methods given on jspsych.org. I'm using the fixed repetitions option. It should repeat every trial 3 times, but in a random order. Currently the repetition is working, but it repeats the trials in order (see shortened in- & output below). The trials are variables defined before. I tried to set randomize_order to true, but that didn't change a thing. 

Probably an easy one, but I could not find a solution by now. Any hint would be appreciated! :)
Also: Is there any option to use the fixed repetitions with unequal probabilities? Adding "weights" is not working with this one.

var timeline_test_random = {
  timeline: [trial1, trial2, trial3, trial1_2, trial2_2, trial3_2],
  sample: {
    type:'fixed-repetitions',
    size: 3    
    }
}

Output:
trial1, trial2, trial3, trial1_2, trial2_2, trial3_2, trial1, trial2, trial3, trial1_2, trial2_2, trial3_2, trial1, trial2, trial3, trial1_2, trial2_2, trial3_2

**Comment by jodeleeuw** [comment, marked as answer]
Hi @fin329,

The `sample` parameter (and `randomize_order`) only applies to the timeline variables, not the timeline itself. The motivation for this is that the timeline represents the fixed procedure that runs (e.g., show a fixation cross, show a stimulus, show feedback) and the timeline variables represent what is different between trials. This is why you are getting the same order each time.

**Reply by fin329** [reply]
Thanks a lot, didn't realize that! :)
